crates/obj/src/patchable.rs

```diff
@@ -0,0 +1,342 @@
+use anyhow::{bail, ensure, Error};
+use object::elf::*;
+use object::endian::LittleEndian;
+use std::ffi::CStr;
+use std::mem::size_of;
+use std::os::raw::c_char;
+
+/// Checks if ELF is supported.
+pub fn ensure_supported_elf_format(bytes: &[u8]) -> Result<(), Error> {
+    parse_elf_object_mut(bytes)?;
+    Ok(())
+}
+
+/// Converts ELF into loadable file.
+pub fn convert_object_elf_to_loadable_file(bytes: &mut Vec<u8>) {
+    // TODO support all platforms, but now don't fix unsupported.
+    let mut file = match parse_elf_object_mut(bytes) {
+        Ok(file) => file,
+        Err(_) => {
+            return;
```

Comment:
This seems like it's a bit of a scary error to omit, could we return an error here instead and use error messages to guide fixing this later?